### ISSUE TITLE
rhcos variant has been removed; use openshift variant instead:  https://coreos.github.io/butane/upgrading-openshift/

### ISSUE THREAD

**Issue by tomazb** [RedHat-EMEA-SSA-Team/hetzner-ocp4#264]
RHEL 8.7 with git pull from today and used "`openshift_version: 4.12.9`" in cluster.yaml.

```
TASK [openshift-4-cluster : Create RHCOS Config] *******************************
changed: [host] => (item=bootstrap)
changed: [host] => (item=master)
changed: [host] => (item=worker)

TASK [openshift-4-cluster : Mangle ignition config] ****************************
failed: [host] (item=bootstrap) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/opt/openshift-client-4.12.9/butane", "--files-dir", "/home/tomaz/hetzner-ocp4/ansible/../ocp1", "--output", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/bootstrap-extra.ign", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/bootstrap.rcc"], "delta": "0:00:00.009118", "end": "2023-04-09 17:32:43.776963", "item": "bootstrap", "msg": "non-zero return code", "rc": 1, "start": "2023-04-09 17:32:43.767845", "stderr": "Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/", "stderr_lines": ["Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/"], "stdout": "", "stdout_lines": []}
failed: [host] (item=master) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/opt/openshift-client-4.12.9/butane", "--files-dir", "/home/tomaz/hetzner-ocp4/ansible/../ocp1", "--output", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/master-extra.ign", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/master.rcc"], "delta": "0:00:00.009943", "end": "2023-04-09 17:32:44.056448", "item": "master", "msg": "non-zero return code", "rc": 1, "start": "2023-04-09 17:32:44.046505", "stderr": "Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/", "stderr_lines": ["Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/"], "stdout": "", "stdout_lines": []}
failed: [host] (item=worker) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/opt/openshift-client-4.12.9/butane", "--files-dir", "/home/tomaz/hetzner-ocp4/ansible/../ocp1", "--output", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/worker-extra.ign", "/home/tomaz/hetzner-ocp4/ansible/../ocp1/worker.rcc"], "delta": "0:00:00.009103", "end": "2023-04-09 17:32:44.335457", "item": "worker", "msg": "non-zero return code", "rc": 1, "start": "2023-04-09 17:32:44.326354", "stderr": "Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/", "stderr_lines": ["Error translating config: rhcos variant has been removed; use openshift variant instead: https://coreos.github.io/butane/upgrading-openshift/"], "stdout": "", "stdout_lines": []}
```

**Comment by avoigtma** [comment]
Looks like the Butane config must be modified, see https://coreos.github.io/butane/upgrading-openshift/#from-rhcos-version-010-to-openshift-version-480.

File: ansible/roles/openshift-4-cluster/tasks/create-ignition.yml

Diff:
45,50c45,46
<       variant: openshift
<       version: 4.8.0
<       metadata:
<         labels:
<           machineconfiguration.openshift.io/role: {{ item }}
<         name: config-openshift-{{ item }}
---
>       variant: rhcos
>       version: 0.1.0

This let's the installer to run thru (still waiting for my installation to complete to confirm above patch is sufficient).

**Comment by rbo** [comment]
@tomazb can you share the version of Butane CLI?
```
$ butane -V
Butane 0.18.0
```

**Comment by rbo** [comment]
Never mind, @avoigtma patch looks good. We have to add `--raw` to butane too because by default it creates a machineconfig and not an ignition. 

https://coreos.github.io/butane/upgrading-openshift/#machineconfig-generation

**Comment by rbo** [comment]
The butane/ignition change is only needed if you run the installation with IPv6. I will remove the steps for an IPv4 only installation.

**Comment by avoigtma** [comment]
Quick fix is to go back to `butane? v0.17, but @rbo thanks for the further comments, did not have time for digging into the butane stuff.

File `ansible/roles/openshift-4-cluster/defaults/main.yml`

```
84c84
< butane_cli_location: "{{ openshift_mirror }}/pub/openshift-v4/clients/butane/v0.17.0/butane-amd64"
---
> butane_cli_location: "{{ openshift_mirror }}/pub/openshift-v4/clients/butane/latest/butane-amd64"

```

Then remove the installed version 
```
rm -fr /opt/openshift-*
rm /usr/local/bin/butane
```
and rerun `setup.yml`playbook.

**Comment by rbo** [comment]
I have a fix available, last test running through. I try to get everything in master branch today.

**Comment by rbo** [comment]
Fix merged with #266 into devel and will be merged into master with #252

**Comment by rbo** [comment]
All merged into master and releases.
